caffe2/onnx/backend.cc

```diff
@@ -381,11 +381,10 @@ Caffe2Ops Caffe2Backend::CreateCast(
   auto c2_op = CommonOnnxNodeToCaffe2Ops(onnx_node, ctx);
 
   auto onnx_dtype =
-      onnx_node->attributes.get<int64_t>("to", TensorProto::UNDEFINED);
-  auto c2_dtype = caffe2::TensorProto::UNDEFINED;
+      onnx_node->attributes.get<int64_t>("to", ::ONNX_NAMESPACE::TensorProto::UNDEFINED);
+  auto c2_dtype = caffe2::TensorProto::FLOAT;
```

Comment:
there's a couple options for not special casing FLOAT.

1) The "modern" thing to do is to leave it unset initially and to rely on MSan to catch if it doesn't get assigned. This catches the problem of accidentally missing to overwrite it with whatever the correct value is.

2) IMO the cleaner thing to do with switch statements that are logically mapping functions is to move them to their own function, e.g.
```
caffe2::TensorProto::DType to_caffe2_dtype(ONNX_NAMESPACE::TensorProto::Type onnx_dtype) {
  switch (onnx_dtype) {
    case ONNX_NAMESPACE::TensorProto::FLOAT:
      return caffe2::TensorProto::FLOAT;
    case ONNX_NAMESPACE::TensorProto::UINT8:
      return caffe2::TensorProto::UINT8;
    case ONNX_NAMESPACE::TensorProto::DOUBLE:
      return caffe2::TensorProto::DOUBLE;
    default:
      CAFFE_THROW(
          "Casting to '",
          onnx_dtype,
          "' dtype is not supported");
  }
}
```

You avoid the risk of `c2_dtype` being unset and it can become `const` in the caller's scope. The `return`s end up being cleaner than the `break`s, IMO and you save a line for each one to boot.